### DISCUSSION TITLE
Fighting With process​.​env​.​NODE_ENV​ ​!==​ ​'production'​

### DISCUSSION THREAD

**Discussion by ghost** [urql-graphql/urql#2385] [Q&A]
I am using Solid Start which is the Coolest Solid.js SSR framework with Cloudflare Workers. As We all know Cloudflare Workers don't support Node.js built-ins like process & in Dev Mode, it works fine but after compiling the Project I find process​.​env​.​NODE_ENV​ ​!==​ ​'production'​ 13 places in the compiled code, which makes deploying to Cloudflare Workers to Fail.

Why is urql/core depends on process​.​env​.​NODE_ENV, because it can run on browsers.

Is there an solution for these? Or I have to Delete process​.​env​.​NODE_ENV​ ​!==​ ​''production' always after compiling the Project.

![Screenshot_20220405-164759_GitHub.jpg](https://user-images.githubusercontent.com/96905008/161839433-068e3932-ba54-4224-a4be-a6719a109884.jpg)

**Comment by JoviDeCroock** [comment]
You can use a replace-plugin in your build pipeline for your worker like https://www.npmjs.com/package/@rollup/plugin-replace

**Reply by ghost** [reply]
Ok, Thanks. I will try to use it & give you my feedback.
Also I would like to tell you that @urql/core is Really Awesome & Keep making it better.

**Reply by ghost** [reply]
Can you tell me why urql is using process​.​env​.​NODE_ENV​ & it's really necessary to be there.

**Reply by JoviDeCroock** [reply]
Most of it are performance and bundle optimizations, using this enables you to warn users of wrong inputs, .... in development, while replacing and minimizing in production removes that code-path making the eventual bundle smaller

**Reply by ghost** [reply]
So If it's removed when bundling, Why It's being included in the output of my project. May be vite is not doing a great job on here. I will contact Solid Start Community for Help.

I wish process​.​env​.​NODE_ENV​ is not included at first. But if it's useful, No Problem.

**Reply by JoviDeCroock** [reply]
Yeah, I guess we should come up with better solutions to achieve this developer-friendly dev bundle and optimized production bundle 😅 I think it might be because vite uses `import.meta.env` rather than `process`

**Reply by mwahlhuetter** [reply]
Has anyone come up with a solution to this? This prevents me from using urql with vite.
